src/server/network.ts

```diff
@@ -326,6 +370,14 @@ export class Response extends SdkObject {
     return this._timing;
   }
 
+  serverIPAddressAndPort(): IPAddressAndPort {
+    return this._serverIPAddressAndPort;
+  }
+
+  securityDetails(): SecurityDetails | undefined {
+    return this._securityDetails;
+  }
```

Comment:
These might need to be change to returning Promises to accom WK's request lifecycle. Thoughts?

Reply:
They shouldn't ever need to be promises. We would instead just delay emitting the response event.

Reply:
Is this the line that is currently triggering the emission of the response event:

https://github.com/microsoft/playwright/blob/475a417dfc29caeb644eb721cbcabd5bb8980636/src/server/webkit/wkPage.ts#L980

I don't think moving that into 

https://github.com/microsoft/playwright/blob/475a417dfc29caeb644eb721cbcabd5bb8980636/src/server/webkit/wkPage.ts#L991

works out but I also am likely misunderstanding the flow of events.

Reply:
Updated per Slack comments. Everything's a Promise!